feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts

```diff
@@ -76,6 +77,38 @@ export class ConfigureProductComponent {
     return this.isConfiguratorTypeReadOnly(configuratorType) ? 'false' : 'true';
   }
 
+  /**
+   * Verifies whether a configurator type of a product contains a read-only postfix and
+   * a product is a base product.
+   *
+   * @param product - product
+   * @returns - If the configurator type contains a read-only postfix and
+   * a product is a base product then returns true, otherwise false.
+   */
+  isReadOnlyBaseProduct(product: Product): boolean {
+    return (
+      this.isConfiguratorTypeReadOnly(product.configuratorType) &&
+      this.isBaseProduct(product)
+    );
+  }
+
+  /**
+   * Verifies if a product is a base product.
+   *
+   * In the catalog the product is a base one when baseProduct and code are defined and equal.
+   * In the product detail page a product is a base one when only code is defined.
+   *
+   * @param product - product
+   * @returns - If a product a base product then returns true, otherwise false.
+   * @protected
+   */
+  protected isBaseProduct(product: Product): boolean {
+    return (
+      product.baseProduct === product.code ||
+      (!product.baseProduct && !!product.code)
+    );
```

Comment:
```suggestion
    return (
      product.baseProduct === product.code || !product.baseProduct 
    );
```
I assume `product.code` may never be `undefined`, as we query the product data using the code.